django/core/signing.py

```diff
@@ -130,17 +116,7 @@ def loads(s, key=None, salt='django.core.signing', serializer=JSONSerializer, ma
 
     The serializer is expected to accept a bytestring.
     """
-    # TimestampSigner.unsign() returns str but base64 and zlib compression
-    # operate on bytes.
-    base64d = TimestampSigner(key, salt=salt).unsign(s, max_age=max_age).encode()
-    decompress = base64d[:1] == b'.'
-    if decompress:
-        # It's compressed; uncompress it first
-        base64d = base64d[1:]
-    data = b64_decode(base64d)
-    if decompress:
-        data = zlib.decompress(data)
-    return serializer().loads(data)
+    return TimestampSigner(key, salt=salt).unsign_object(s, serializer=serializer, max_age=max_age)
 
```

Comment:
Emptying `signing.dumps` and `loads` opens up the possibility of replacing their calls with a call to `sign_object` and `unsign_object` in a future refactoring. I see that they are called in `contrib.sessions.backends.signed_cookies` and `base`.

Reply:
Jupp, this is also the reason why I have put the move into a separate PR as well https://github.com/django/django/pull/13792 (this one builds upon that foundation).